tests/test-helpers.js

```diff
@@ -36,7 +36,76 @@ const checkMaybeFunction = (factory, dependencies, identifier = MAYBE_FUNCTION)
     `;
 };
 
+const checkVarArgsResult = (
```

Comment:
Just to make this function name more descriptive, can we please change to `checkVariableDepAndFactoryResult` or similar.

Also similar to a previous comment, let's make the arguments an object instead, i.e. `myFunc({ factory, dependencies, ... })`, for increased readability at call sites and ease of future refactors.